lib/dap/filter/ldap.rb

```diff
@@ -36,9 +36,11 @@ def decode(data)
       begin
         elem_decoded = OpenSSL::ASN1.decode(element)
       rescue Exception => e
-        err_msg = 'FilterDecodeLdapSearchResult - Unable to decode ANS1 element'
+        err_msg = 'FilterDecodeLdapSearchResult - Unable to decode ANS.1 element'
         $stderr.puts "#{err_msg}: #{e}"
         $stderr.puts e.backtrace
+        $stderr.puts "\nElement:\n#{element}\n"
+        $stderr.puts "\nElement hex:\n#{element.unpack('H*')}\n\n"
```

Comment:
It looks like you are sprinkling newlines here to add with readability of the error logs when there are lots of errors.  You might want to try not adding additional explicit leading/trailing newlines, as `puts ` already adds one, and if you want to add a newline in the middle of a message to aid in readability, I'd suggest adding a tab after the newline.  IOW, this:

```
        $stderr.puts "Element:\n\t#{element}"
        $stderr.puts "Element hex:\n\t#{element.unpack('H*')}"
```

Otherwise you end up with lots of blank lines.